src/WebJobs.Script.WebHost/BreakingChangeAnalysis/ChangeAnalysisService.cs

```diff
@@ -0,0 +1,158 @@
+        {
+            _analysisTask = Task.Delay(TimeSpan.FromMinutes(1), _cancellationTokenSource.Token)
+                .ContinueWith(async t =>
+                {
+                    if (!_cancellationTokenSource.IsCancellationRequested && _hostStateProvider.IsPrimary)
+                    {
+                        await TryLogBreakingChangeReportAsync(_cancellationTokenSource.Token);
+                    }
+                })
+                .Unwrap();
+        }
+
+        internal async Task TryLogBreakingChangeReportAsync(CancellationToken cancellationToken)
+        {
+            bool lockAcquired = false;
+            try
+            {
+                lockAcquired = await _analysisSemaphore.WaitAsync(30000, cancellationToken);
+
+                ChangeAnalysisState analysisState = await _changeAnalysisStateProvider.GetCurrentAsync(cancellationToken);
+
+                // Currently, only performing analysis if we haven't done so within the previous 7 days
+                if (analysisState.LastAnalysisTime > DateTimeOffset.UtcNow.AddDays(-7))
+                {
+                    _logger.LogInformation("Skipping breaking change analysis.");
+                    return;
+                }
+
+                _logger.LogInformation("Initiating breaking change analysis...");
+
+                LogBreakingChangeReport(cancellationToken);
+
+                await _changeAnalysisStateProvider.SetTimestampAsync(DateTimeOffset.UtcNow, analysisState.Handle, cancellationToken);
+
+                _logger.LogInformation("Breaking change analysis operation completed.");
+            }
+            catch (OperationCanceledException)
+            {
+                _logger.LogInformation("Breaking change analysis operation cancelled.");
+            }
+            catch (Exception exc)
+            {
+                _logger.LogWarning(exc, "Breaking change analysis operation failed");
+            }
+            finally
+            {
+                if (lockAcquired)
+                {
+                    _analysisSemaphore.Release();
+                }
+            }
+        }
+
+        public IEnumerable<AssemblyReport> LogBreakingChangeReport(CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var generator = new BreakingChangeReportGenerator();
+
+            IEnumerable<Assembly> functionContextAssemblies = AppDomain.CurrentDomain.GetAssemblies()
+                .Where(a => AssemblyLoadContext.GetLoadContext(a) == FunctionAssemblyLoadContext.Shared);
+
+            var reports = new List<AssemblyReport>();
+            foreach (var assembly in functionContextAssemblies)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                AssemblyReport report = generator.ProduceReport(new Uri(assembly.CodeBase).LocalPath);
+                _logger.LogInformation(JsonConvert.SerializeObject(report));
```

Comment:
Do we want this as Information?

Reply:
Are you thinking this is too verbose?

Reply:
Just wonder if we'll get a bunch of questions about it when it shows up in customer's App Insights. Diagnostic stuff like that I'd log as Debug.

Reply:
Makes sense. Will change.